### ISSUE TITLE
Info about emo_detector_cli

### ISSUE THREAD

**Issue by mastiffs92** [luca-m/emotime#65]
Hi ,
what is the image file path?
I don't understand when I launch the script python that passes as "image file path"?
![screenshot](https://cloud.githubusercontent.com/assets/15649307/12477673/1d4de504-c031-11e5-9963-4737b0fcb7a2.png)
Thanks in advance for thw answer!

**Comment by egottardo** [comment]
Hi,
The "image file path" refers to the location where the original images for training is located.

If you are using Extended Cohn-Kanade database there are four folders:
1) The Images (extended-cohn-kanade-images.zip) -  there are 593 sequences across 123 subjects which are FACS coded at the peak frame. All sequences are from the neutral face to the peak expression.
2) The Landmarks (Landmarks.zip) - All sequences are AAM tracked with 68points landmarks for each image. 
3) The FACS coded files (FACS_labels.zip) - for each sequence (593) there is only 1 FACS file, which is the last frame (the peak frame). Each line of the file corresponds to a specific AU and then the intensity. An example is given below.
4) The Emotion coded files (Emotion_labels.zip) - ONLY 327 of the 593 sequences have emotion sequences. 

Then, 'image file path' refers to the first (1)  and 'emotion folder' refers to fouth (4).

Hope this helps.
Ernani Gottardo

**Comment by mastiffs92** [comment]
I thank you for the answer,
you mean this folder
![screenshot](https://cloud.githubusercontent.com/assets/15649307/12482541/7f3de7cc-c050-11e5-8584-1b42a5205261.png)
but I will pass this path does not work
![screenshot1](https://cloud.githubusercontent.com/assets/15649307/12482595/cd72c958-c050-11e5-9e6e-5c5e2b478891.png)

**Comment by mastiffs92** [comment]
When to Use python script everything works in the testing phase, only that I don't see how files .cpp are recalled. I'm very interested for educational purposes.

**Comment by egottardo** [comment]
Excuse-me. I didn't observe that problem isn't in training phase (when creating new classification model). You are alread executing emo_detector_cli. In this case you should inform the path of one image.
Below a figure of one test.
![captura de tela de 2016-01-21 15 28 13](https://cloud.githubusercontent.com/assets/16702452/12489695/b1cb6842-c058-11e5-8958-07d6a4f6c152.png)

**Comment by mastiffs92** [comment]
Thanks for the answer, now I will try again !

**Comment by mastiffs92** [comment]
where am I wrong?
![screenshot2](https://cloud.githubusercontent.com/assets/15649307/12509902/401fc02c-c106-11e5-9734-a4a8ddd46523.png)

**Comment by egottardo** [comment]
It seems that the face was not locate by cbcl1 in the image. I'm also having trouble with cbcl1 in locating faces. 
I suggest that you test with another images or with haarcascade_frontalface_default.xml / haarcascade_frontalface_alt2.xml

**Comment by mastiffs92** [comment]
I have tested with your changes but situation isn't change.
![screenshot3](https://cloud.githubusercontent.com/assets/15649307/12511326/d54ba382-c110-11e5-8ed7-c870ba6d940e.png)

**Comment by mastiffs92** [comment]
No idea? @egottardo

**Comment by dbellavista** [comment]
Try by adding some debug prints in facepreprocessor, to understand why it
fails. Keep in mind also that you face has an hat. This can be a problem

**Comment by mastiffs92** [comment]
ok I will try , thanks.

**Comment by egottardo** [comment]
Its strange! For me it's working fine, but i'm using the model with 1vsallext.
I suggest that you try with a model  trained with 1vsallext.

**Comment by mastiffs92** [comment]
OK Thanks! I will try!

**Comment by r-sreesaran** [comment]
even i am getting the same error while using 1vsallext, Any ways too solve this problem

**Comment by mastiffs92** [comment]
@r-sreesaran Also with models 1_vs _all_ext you have same problems?

**Comment by r-sreesaran** [comment]
@mastiffs92  the error was happening because i was using .jpg file but now it throws up a error 

OpenCV Error: Bad argument (The SVM should be trained first) in CvSVM::predict, file /root/essentials/opencv/opencv-2.4.9/modules/ml/src/svm.cpp, line 2131
terminate called after throwing an instance of 'cv::Exception'
  what():  /root/essentials/opencv/opencv-2.4.9/modules/ml/src/svm.cpp:2131: error: (-5) The SVM should be trained first in function CvSVM::predict

**Comment by mastiffs92** [comment]
I use the format of image file path '.png' and also i am using the compiled version of binaries for linux.

**Comment by husnain-ahmad** [comment]
I am sorry but I am getting the following error. If you can kindly guide me out of it, that would be so awesome.
What am I doing wrong? This is happening with both the master and the experimental version. I have also tried changing the parameters. I am using CK extended database files as the input image.
```
dell@gandalf:~/Desktop/emotime/emotime-1.2-experimental/assets$ ./emo_detector_cli svm ../resources/haarcascade_frontalface_default.xml ../resources/haarcascade_eye.xml 53 53 1 5 8 ../../svm_1vsallext_1_5_8_95c2eb0b58/*.xml
Insert the image file path: 
/home/dell/Downloads/cohn-kanade-images/S045/001/S045_001_00000001.png
Processing '/home/dell/Downloads/cohn-kanade-images/S045/001/S045_001_00000001.png'
OpenCV Error: Sizes of input arguments do not match (The sample size is different from what has been used for training) in cvPreparePredictData, file /home/dell/OpenCV/modules/ml/src/inner_functions.cpp, line 1114
terminate called after throwing an instance of 'cv::Exception'
  what():  /home/dell/OpenCV/modules/ml/src/inner_functions.cpp:1114: error: (-209) The sample size is different from what has been used for training in function cvPreparePredictData

Aborted (core dumped)
```

**Comment by egottardo** [comment]
In my tests, this error occured when the parameters WIDTH HEIGHT NWIDTHS NLAMBDAS NTHETAS (in your case 53 53 1 5 8) are different from the trainning phase.

**Comment by husnain-ahmad** [comment]
@egottardo Yes you are right. I got it working after choosing the exact same parameter values for which the classifiers were trained on.
Thank you for your support.

**Comment by rajat70595** [comment]
@egottardo  @husnain-ahmad  I tried with the same parameters as in the configuration file used to train the system. Following is the command I used : 
./emo_detector_cli svm ../resources/haarcascade_frontalface_default.xml ../resources/haarcascade_eye.xml 32 32 1 5 6 /home/rajat/Downloads/emotime-master/dataset/classifiers/svm/*.xml

Insert the image file path: 
/home/rajat/Desktop/My_Project/google_004.jpg

But here is the error I am getting : 
Processing '/home/rajat/Desktop/My_Project/google_004.jpg'
OpenCV Error: Assertion failed (dsize.area() || (inv_scale_x > 0 && inv_scale_y > 0)) in resize, file /build/buildd/opencv-2.4.8+dfsg1/modules/imgproc/src/imgwarp.cpp, line 1825
terminate called after throwing an instance of 'cv::Exception'
  what():  /build/buildd/opencv-2.4.8+dfsg1/modules/imgproc/src/imgwarp.cpp:1825: error: (-215) dsize.area() || (inv_scale_x > 0 && inv_scale_y > 0) in function resize

Aborted

*Please help*

**Comment by danielmihai** [comment]
try scaling the google image(s) to the resolution CK+ images have :))

**Comment by egottardo** [comment]
I found the same problem with some small images. Since I did not have time to investigate the problem i temporarily manually increased the images.
